scrape/scrape.go

```diff
@@ -955,11 +958,13 @@ type scrapeCache struct {
 
 	// Parsed string to an entry with information about the actual label set
 	// and its storage reference.
-	series map[string]*cacheEntry
+	series    map[string]*cacheEntry
+	seriesSHA map[[sha512.Size256]byte]*cacheEntry
```

Comment:
(No good place to put this comment otherwise, sorry).

The logic in `iterDone` is no longer correct, since it only operates on `series` and `droppedSeries`.

Reply:
Good catch!